### PR TITLE
seabios_strict: add strict option on/off test

### PR DESCRIPTION
**Pull request by ybduan**
Check boot result when strict is on/off

ID: 1469040

Signed-off-by: ybduan <yduan@redhat.com>

### CODE REVIEW

**Comment by vivianQizhu** [comment]
@xiangchunfu Please help review, thanks.

**Comment by xiangchunfu** [comment]
linux guest:
 (1/2) Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.seabios_strict.on: PASS (459.16 s)
 (2/2) Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.seabios_strict.off: PASS (102.04 s)

windows guest:
 (1/4) Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.i386.sp1.io-github-autotest-qemu.seabios_strict.on: PASS (459.75 s)
 (2/4) Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.i386.sp1.io-github-autotest-qemu.seabios_strict.off: PASS (101.71 s)
 (3/4) Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.seabios_strict.on: PASS (459.64 s)
 (4/4) Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.seabios_strict.off: PASS (102.38 s)

so, remove "linux only"

**Comment by ybduan** [comment]
Updated, thanks! @xiangchunfu

File: qemu/tests/seabios_strict.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+        if utils_misc.wait_for(boot_check_extra, timeout, 1):
+            if boot_strict:
+                err = "Guest tries to boot from DVD/CD when 'strict' is on"
+                test.fail(err)
+        else:
+            if not boot_strict:
+                err = "Guest does not try to boot from DVD/CD"
+                err += " when 'strict' is off"
+                test.fail(err)
```
Comment (xiangchunfu) [inline qemu/tests/seabios_strict.py]: @ybduan If strict is off, should try to boot from DVD

Comment (xiangchunfu) [reply qemu/tests/seabios_strict.py]: @ybduan ignore this comment above.  It makes sense.

**Comment by xiangchunfu** [comment]
For windows guest. 
(1/4) Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.i386.sp1.io-github-autotest-qemu.seabios_strict.on: PASS (459.95 s)
 (2/4) Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.i386.sp1.io-github-autotest-qemu.seabios_strict.off: PASS (102.47 s)
 (3/4) Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.seabios_strict.on: PASS (460.36 s)
 (4/4) Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.seabios_strict.off: PASS (102.33 s)

For linux guest.
 (1/2) Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.seabios_strict.on: PASS (459.37 s)
 (2/2) Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.seabios_strict.off: PASS (101.96 s)

**Comment by xiangchunfu** [comment]
Acked-by: Xiangchun Fu  xfu@redhat.com

File: qemu/tests/seabios_strict.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    def cleanup_cdrom():
+        """
+        Remove 'test' cdrom
+        """
+        logging.info("cleaning up test cdrom")
+        cdrom_test = utils_misc.get_path(data_dir.get_data_dir(),
+                                         params.get("cdrom_test"))
+        os.remove(cdrom_test)
+
+    def get_output(session_obj):
```
Comment (vivianQizhu) [inline qemu/tests/seabios_strict.py]: This type of one line function is not necessary.

File: qemu/tests/seabios_strict.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        """
+        boot_fail_infos check
+        """
+        return re.search(fail_infos, get_output(vm.serial_console), re.S)
+
+    def boot_check_extra():
+        """
+        boot_fail_infos_extra check
+        """
+        return re.search(fail_infos_extra, get_output(vm.serial_console), re.S)
```
Comment (vivianQizhu) [inline qemu/tests/seabios_strict.py]: `boot_check()` and `boot_check_extra()` can be merged, and even can be dropped, you can use `lamda` in utils_misc.wait_for().

Comment (vivianQizhu) [reply qemu/tests/seabios_strict.py]: @ybduan I don't understand why it is complex, they are 1. only one sentence and 2. totally same just with a variant change. Just simply replace the function name in your wait_for() with the actually executed line, with an extra `lambda: ` in front of it. Or at least it should be one single function with a parameter like `file_info`.
Please do avoid those one sentence functions in your code in the future, as well as redundant implementations. Thanks.

Comment (vivianQizhu) [reply qemu/tests/seabios_strict.py]: @ybduan For 1, maybe, but not this one. And not most ones. Anyway I don't accept that without sufficient reason.
2 Acceptable to me. But for 3, the existence of _foo() does not make sense at all. Thanks.

Comment (vivianQizhu) [reply qemu/tests/seabios_strict.py]: @ybduan By the way, I once got other maintainers complain about me accepting one of your previous pull request which has similar one sentence implementation, and I absolutely agree with that complain. So let's stop argue this please. Thank you.

Comment (vivianQizhu) [reply qemu/tests/seabios_strict.py]: @ybduan I understand that directly use `re` in wait_for() will make lines too long which might annoy you, so I would accept the third solution if you remove the unreasonable `_foo()`, so directly
```
def check(info):
    return re.search(info, vm.serial_console.get_stripped_output(), re.S)
```
Thanks.

Comment (ybduan) [reply qemu/tests/seabios_strict.py]: If we remove it as you wish:

17:34:53 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_plugins_vt-65.0-py2.7.egg/avocado_vt/test.py:435
17:34:53 ERROR| Traceback (most recent call last):                              
17:34:53 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-65.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
17:34:53 ERROR|     return fn(*args, **kwargs)                                  
17:34:53 ERROR|   File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/seabios_strict.py", line 92, in run
17:34:53 ERROR|     if not utils_misc.wait_for(check(fail_infos), timeout, 1):  
17:34:53 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-65.0-py2.7.egg/virttest/utils_misc.py", line 685, in wait_for
17:34:53 ERROR|     output = func()                                             
17:34:53 ERROR| TypeError: 'NoneType' object is not callable

Comment (vivianQizhu) [reply qemu/tests/seabios_strict.py]: @ybduan You need a `lamda:` in front of `check(fail_infos)`

Comment (ybduan) [reply qemu/tests/seabios_strict.py]: Updated

**Pull request merged**